js/src/bootstrap/rm.js

```diff
@@ -0,0 +1,76 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const { getDescribe, getIt } = require('../utils/mocha')
+
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const invalidArg = 'this/Is/So/Invalid/'
+
+module.exports = (createCommon, options) => {
+  const describe = getDescribe(options)
+  const it = getIt(options)
+  const common = createCommon()
```

Comment:
Why does common need to created here, if there are no options for it?

Reply:
We need to create a `common` object for each command now that they have been split into modules since it tracks nodes that are created so that they can be destroyed when teardown is called.

```js
// Before:
tests.files(common)
// -> calls:
tests.files.add(common) // teardown destroys nodes created for files.add
tests.files.get(common) // teardown destroys nodes created for files.add AND files.get - ruh roh!
// etc.

// Now:
tests.files(createCommon)
// -> calls:
tests.files.add(createCommon) // teardown destroys nodes created for files.add
tests.files.get(createCommon) // teardown destroys nodes created for files.get
```

The alternative would be to make the `common` object smarter - to remove nodes from it's `nodes` array when teardown is called. However I was thinking that if we were to ever run these suites concurrently (please! one day!) then things would break.

From the PR description:

> ~~This work is backwards compatible with the previous API so no changes to dependent code needs to be made.~~ I wanted this to be backwards compatible, but quickly ran into an issue where the `common` object we pass to suites expects to only be used once. I started to pass it to multiple suites and the `teardown` function was trying to tear down nodes that were already stopped! So, unfortunately **breaking change**: `common` should now be a function that creates a `common` object for use with the tests. I'll submit PR's to js-ipfs (https://github.com/ipfs/js-ipfs/pull/1389) and js-ipfs-api (https://github.com/ipfs/js-ipfs-api/pull/785) to deal with this.